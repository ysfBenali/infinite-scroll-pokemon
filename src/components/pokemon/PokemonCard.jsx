import { useEffect, useState } from "react";
import { API_IMG_baseURL } from "../../helpers/utils/constants";
import { CardContainer, CardIndex, Avatar } from "../styles";

const PokemonCard = ({ name, url, handleClick }) => {
  const [pokemon, setPokemon] = useState({
    name: "",
    url: "",
    index: "",
    imageUrl: "",
  });

  useEffect(() => {
    let index = url.split("/")[url.split("/").length - 2];
    let imageUrl = `${API_IMG_baseURL}/${index}.svg`;
    setPokemon({ name, url, index, imageUrl });
  }, []);

  return (
    <CardContainer onClick={handleClick} data-testid='pokemonId'>
      <CardIndex>
        <span>#</span>
        {pokemon.index}
      </CardIndex>
      <Avatar src={pokemon.imageUrl} alt={pokemon.name} />
      <p>{pokemon.name}</p>
    </CardContainer>
  );
};

export default PokemonCard;
