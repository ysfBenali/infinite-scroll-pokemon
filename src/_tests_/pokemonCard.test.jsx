import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer'
import PokemonCard from "../components/pokemon/PokemonCard"

const pokemon = {
    name: "bulbasaur",
    url: "https://pokeapi.co/api/v2/pokemon/1/",
    index: "1",
  };

it("pokemon item render with name", () => {
    const {getByTestId,getByText}= render(<PokemonCard name={pokemon.name} url={pokemon.url}/>)
    expect(getByTestId("pokemonId")).toBeInTheDocument()
    expect(getByText(pokemon.name)).toBeInTheDocument()
})

it('match snapshot', () => {
    const tree = renderer.create(<PokemonCard name={pokemon.name} url={pokemon.url}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });