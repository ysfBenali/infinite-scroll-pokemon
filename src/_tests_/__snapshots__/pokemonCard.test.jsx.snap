// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="sc-dkPtyc cdXur"
  data-testid="pokemonId"
>
  <span
    className="sc-hKwCoD jlkegf"
  >
    <span>
      #
    </span>
    
  </span>
  <img
    alt=""
    className="sc-eCImvq erRmYc"
    src=""
  />
  <p>
    
  </p>
</div>
`;
